web-console/knikubevirt/Create-vm/step-2-networking/create-vm-networking.md

```diff
@@ -3,17 +3,20 @@
 ## Step 2: Networking
 
 ![Default](img/3-00.png)
-The user enters the Networking step with an existing NIC pre-defined, set to Type (local). In this step, the 'Review and Create' button is still available to skip the rest of the steps. At this point, the user can select to customize this NIC, add new ones, or remove any they wish.
+The user enters the Networking step with an existing  Network interface pre-defined, set to Type (local). In this step, the 'Review and Create' button is still available to skip the rest of the steps. At this point, the user can select to customize this Network interface, add new ones, or remove any they wish.
 
 ![Adding NIC](img/3-01.png)
-Clicking on 'Add NIC' will pop up a "Create NIC" modal. The NIC name is prepopulated with a numeric count form of existing NICs on this flow. The user may change the NIC name. All the fields are preselected except Type. In order for the user to apply and add this NIC, the user must select one of the type options.
+Clicking on 'Add  Network interface' will pop up a "Add Network interface" modal. The Network interface name is prepopulated with a numeric count form of existing Network interfaces on this flow but the user may change it. In order for the user to apply and add this Network interface, they must select one of the type options.
+In case the user selects 'Masquerade' on the Type field, they can add multiple Ports.
+In that case the user will get a list of Ports (port name, port number, protocol) which will be then forwarded and exposed to the virtual machine. The ports will be displayed in rows so the user can add or remove them.
 
-![Type - Masquerade](img/3-02.png)
-In case the user selects Masquerade on the Type field, new indented fields will appear.
+![masquerade is selected](img/3-03.png)
 
-![NIC added](img/3-10.png)
-The new NIC is added at the bottom of the list.
-Now the user can edit or remove it.
+The new  Network interface is added at the bottom of the list.
+Clicking the kebab allows the user to edit or remove it.
+
+![NIC added to the table](img/3-10.png)
+
+On removing all Network interfaces, the Networking step enters an empty state.
 
 ![Empty state](img/3-20.png)
```

Comment:
HI @yfrimanm and @matthewcarleton I'm wondering for this state if it would be a better fit to use the link button with the "+" as seen here: https://www.patternfly.org/v4/documentation/react/components/button/

The reason I'm asking is because we already have one secondary button at play on this screen, and I'm wondering if providing the link button with a "+" would help distinguish the action? Please let me know if there is a greater context here that I am not understanding - I'm learning as I go. Thanks!

Reply:
+1 I think that would be a good idea. 
For this empty state in general I'd suggest we drop the grey background, it's not part of the current empty state in PF. I wonder if we need that icon @rachael-phillips how important is it that we stick to what PF suggests (just a link button without the icon).

Reply:
+1

Reply:
![image](https://user-images.githubusercontent.com/44577420/64353967-4277b280-d007-11e9-96fa-4a4021aa85ec.png)

Reply:
> +1, but maybe the '+' will distinguish it from the 'Cancel'? 

I agree that the + helps distinguish the action. @matthewcarleton in this context, it's not that important - with either usage we're in line with a PF standard, so that's not a big concern. I'm only raising it as an option to distinguish it from the other actions on the page. Also, this isn't standardized with PF, but @yfrimanm  I liked your idea of including the "Add Network interface" in the empty state :)

Reply:
yes, thank you! I think it looks much better now.

Reply:
this looks good, although I'd suggest keeping the link in the empty state and not in the footer. We want footer actions to stay consistent and the empty state offers the option to have a link.

Reply:
Yes, consistency is the key :) 
Thanks!
Btw, "Network should not be capitalized in the empty text" - wouldn't that be unconsistent with the other screens?

Reply:
I'm talking about the empty state text, so it would look like "No network interface added" which is sentence case.